collections-arrays/ConvertHexStringToByteArray/src/ConvertHexStringToByteArrayLibrary/ConversionHelpers.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace ConvertHexStringToByteArray.Library;
+
+public static class ConversionHelpers
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int PerformModularArithmeticCalculation(char value) => (value % 32 + 9) % 25;
+
+    public static byte[] FromHexWithModularArithmetic(ReadOnlySpan<char> input)
+    {
+        if (input.StartsWith("0x"))
+            input = input[2..];
+
+        if (input.IsEmpty)
```

Comment:
This is intentional and by design. The first thing is checking for `"0x"`, and stripping it off. If it is empty at that point, then we can return an empty string. It basically allowing (as I state in the article) to handle the case of an empty hex string still be prefaced with the hex marker.

Reply:
@AndreCarvalho - exactly, basically the first one has to be first as the others all depend upon the potential modification. While technically speaking, since the change would be 0 % 2, one could argue that the third check could potentially come first, but I feel like it reads more logically this way. First, check for the "hex marker" and strip it off if necessary. After that is done, check for the empty case and return an empty array. IFF all other checks are clear, we then check for the proper string length.